### ISSUE TITLE
Dartdoc generation failed for specific package (due to size)

### ISSUE THREAD

**Issue by rootext** [dart-lang/pub-dev#5122]
URL: https://pub.dev/help

Dartdoc generation failed for https://pub.dev/packages/opengl package. Version 0.5.0

Exception:
```
Running dartdoc:
Running: pub_dartdoc --input /tmp/pub-dartlang-dartdocITVMUU/pkg --output /tmp/pub-dartlang-dartdocITVMUU/output/opengl/0.5.0 --rel-canonical-prefix https://pub.dev/documentation/opengl/latest --no-validate-links --sdk-dir /tool/stable/dart-sdk
STDOUT:
[2021-09-20T22:57:57.477218] Starting...
Documenting opengl...
Initialized dartdoc with 47 libraries in 35.5 seconds
Generating docs for library opengl from package:opengl/opengl.dart...


STDERR:
Unhandled exception:
DocumentationTooBigException: Reached 2147483648 bytes in the output directory.
#0      PubResourceProvider._aboutToWriteBytes (package:pub_dartdoc/src/pub_hooks.dart:76:7)
#1      _File.writeAsBytesSync (package:pub_dartdoc/src/pub_hooks.dart:179:15)
#2      _File.writeAsStringSync (package:pub_dartdoc/src/pub_hooks.dart:185:5)
#3      DartdocFileWriter.write (package:dartdoc/dartdoc.dart:86:10)
#4      DartdocGeneratorBackend.write (package:dartdoc/src/generator/dartdoc_generator_backend.dart:99:12)
#5      DartdocGeneratorBackend.generateTypeDef (package:dartdoc/src/generator/dartdoc_generator_backend.dart:254:5)
#6      GeneratorFrontEnd._generateDocs (package:dartdoc/src/generator/generator_frontend.dart:269:29)
#7      GeneratorFrontEnd.generate (package:dartdoc/src/generator/generator_frontend.dart:21:5)
#8      Dartdoc.generateDocsBase (package:dartdoc/dartdoc.dart:213:21)
<asynchronous suspension>
#9      Dartdoc.generateDocs (package:dartdoc/dartdoc.dart:249:28)
<asynchronous suspension>
#10     main (file:///project/pkg/pub_dartdoc/bin/pub_dartdoc.dart:33:19)
<asynchronous suspension>
```

Log https://pub.dev/documentation/opengl/latest/log.txt

Every generated page is 0.5 MB and in total all docs are 2.7 GB. It's too big for a such simple API. I believe the list with all members on left side should be optional (except package page) and disabled by default.

**Comment by isoos** [comment]
It looks like https://gitlab.com/ext/dart-opengl/-/blob/master/lib/src/opengl_header.dart is an unusually long file, and for some reason `dartdoc` generates an output that is larger than 2GB for it. Let's raise this issue with dartdoc, as `pub.dev` will limit documentation at this rather large size.

**Comment by isoos** [comment]
It looks like that my initial hunch of having only a single file generated is not really feasible, as it will break permalinks. We are looking to fix this in the long term (see #4807 for future progress on making these files smaller and non-repetitive), but for now, maybe the best solution would be to annotate the file with `@nodoc` tag and skip generating the member documentation.

**Comment by isoos** [comment]
Closing, as the right solution will be the size reduction through #4807, and this issue is a duplicate.
